### DISCUSSION TITLE
Explosion during collapse?

### DISCUSSION THREAD

**Discussion by jacobosaa** [Anura3D/Anura3D_OpenSource#157] [General]
Hello everybody!
I'm currently running a series of simulations of column collapses, using fully saturated granular material and liquid material. While using the first one, in some of the simulations (mainly in those with lighter solid content/density, even using the tutorial material data for that example), the collapses stops and begins to explode. Why does that happen? is it because i'm not using some configuration in the calculation data for the liquid pressure? or is it maybe because my material info is wrong.

![imagen columna colapso final](https://github.com/Anura3D/Anura3D_OpenSource/assets/128545000/473708cd-6e11-4335-a154-5f5151299b7c)
![imagen columna colapso t=6](https://github.com/Anura3D/Anura3D_OpenSource/assets/128545000/7b2f0648-0e2f-4066-ab98-45b6a64cf067)
![imagen columna colapso t=12](https://github.com/Anura3D/Anura3D_OpenSource/assets/128545000/376554fb-ce97-45f1-8f9a-9566b2d2a697)
![imagen columna pre colapso](https://github.com/Anura3D/Anura3D_OpenSource/assets/128545000/69f19378-ffcc-40a4-a952-d73faf64e6eb)

Hoping to read from you soon
Regards from Chile!

**Comment by FrancescaCeccato** [comment]
If the material is saturated you must certainly define alla the properties for solid and liquid. There might be issue with the satability of the simulation, especially for low permeabilities, and this depends on the discretization (mesh and number of material points). There are definitely too many emptu elements in your simulation, you can try with larger number of material points per element, a structured mesh... many different options may improve your results.

**Reply by jacobosaa** [reply]
Hi @FrancescaCeccato , thanks for your reply. Unfortunately, I've tried several configurations trying to avoid this "explosion" phenomena, but it keeps happening. I think that it's due to the physics of the scheme, because, it's the water pressure during the collapse that causes the explosion, and it also happens on different model that I've run. It's  like a chain reaction when some inter-particle collision occurs inside the column due to the collapse and it propagates.

Is there a way to "release" that water pressure in the calculation configuration? I've also reduced the size of the mesh and increased the number of material points per cell. I don't know what else I can do.

Hoping to read from you soon.
Best regards from chile!

**Reply by FrancescaCeccato** [reply]
Yes this could be. You can limit the pore pressure by setting a threshold in the subroutine GetFluidPressure. You may try to avoid negative pressure first, I think this can stabilize the results.

**Reply by jacobosaa** [reply]
Hi Francesca! Thanks for your reply, can you please explain to me how to set that threshold? and the negative pressure step too? or maybe where can I find that information? 

thanks!

**Reply by jacobosaa** [reply]
I must add that due to the soil parameters, I'm currently dealing with high porosity saturated soils. Does it affect?

**Reply by FrancescaCeccato** [reply]
Yes, High porosity may be an issue.
There are several ways to add a threshold, it depends on your problem and the control you want on the parameter.
The basic idea is to add a few lines after the calculation of DSigWP in GetFluidPressure in order to limit the pore pressure increment, or you can add a few lines to control the final pressure in AssignWatandgasPressureToGlobalArray. 
For example you can see that in AssignWatandgasPressureToGlobalArray there is already a restriction for the gas pressure (3phase formulation), but you can adjust accordingly for the water pressure.
